### PR TITLE
stm32 and kinetis adc driver move from ISR (hrt_call_every) to new WQ

### PR DESCRIPTION
**Pull request by dagar**
@davids5 Not the bigger ADC cleanup I was hoping for, but here's a simple quick change to at least get the ADC sampling (and system_power floating point work) out of HRT.

### CODE REVIEW

**Comment by dagar** [comment]
The other minor changes are to bring stm32 and kinetis into sync so that we can factor out the large common portion next.

File: src/drivers/stm32/adc/adc.cpp [hunk 1]
```diff
@@ -455,30 +428,21 @@ ADC::update_system_power(hrt_abstime now)
...
-	} else {
-		_to_system_power = orb_advertise(ORB_ID(system_power), &system_power);
-	}
+	_to_system_power.publish(system_power);
 
 #endif // BOARD_ADC_USB_CONNECTED
 }
 
 uint16_t board_adc_sample(unsigned channel)
 {
```
Comment (davids5) [inline src/drivers/stm32/adc/adc.cpp]: @dagar I think this needs to be re-thought from an atomic point of view.

Comment (dagar) [reply src/drivers/stm32/adc/adc.cpp]: Which part? Are you thinking about the conversion per sample or getting multiple samples (+ system_power) gpios together?

File: src/drivers/kinetis/adc/adc.cpp [hunk 2]
```diff
@@ -207,72 +185,86 @@ ADC::~ADC()
...
 
-int
-ADC::init()
+int board_adc_init()
 {
-	/* Input is Buss Clock 56 Mhz We will use /8 for 7 Mhz */
+	static bool once = false;
 
-	irqstate_t flags = px4_enter_critical_section();
+	if (!once) {
```
Comment (davids5) [inline src/drivers/kinetis/adc/adc.cpp]: This is Not a once.  It is not correct if it does not set once true

This HW does not support versioning and does not need the once.

Comment (dagar) [reply src/drivers/kinetis/adc/adc.cpp]: Dropped.

File: src/drivers/stm32/adc/adc.cpp [hunk 3]
```diff
@@ -467,44 +442,39 @@ ADC::update_system_power(hrt_abstime now)
 
 	/* The valid signals (HW dependent) are associated with each brick */
 #if !defined(BOARD_NUMBER_DIGITAL_BRICKS)
+	irqstate_t flags = px4_enter_critical_section();
```
Comment (davids5) [inline src/drivers/stm32/adc/adc.cpp]: Why?

Comment (dagar) [reply src/drivers/stm32/adc/adc.cpp]: A guess while waiting for a response above. https://github.com/PX4/Firmware/pull/12638#discussion_r310955760

**Comment by dagar** [comment]
No change in `adc test` results relative to master.

**Pull request merged**